app/Services/RatingsService.php

```diff
@@ -11,8 +11,12 @@ public function updateRatings($user, $questions, $church_id, $survey,  $data)
   {
     foreach($questions as $question)
     {
-      $church_question = ChurchQuestion::firstWhere(['question_id' => $question->id, 'church_id' => $church_id, 'survey_id' => $survey->id]);
-      Rating::create(['user_id' => $user->id, 'church_question_id' => $church_question->id, 'score' => $data[$question->id]]);
+      $church_question = ChurchQuestion::firstOrCreate(['question_id' => $question->id, 'church_id' => $church_id, 'survey_id' => $survey->id]);
```

Comment:
Since this line is slightly longer, I would prefer breaking it down for readability & easier to debug.

```php
$church_question = ChurchQuestion::firstOrCreate([
  'question_id' => $question->id,
  'church_id' => $church_id,
  'survey_id' => $survey->id
]);
```